vendor/k8s.io/kubernetes/pkg/volume/glusterfs/glusterfs.go

```diff
@@ -476,9 +484,9 @@ func (r *glusterfsVolumeProvisioner) Provision() (*api.PersistentVolume, error)
 		return nil, err
 	}
 	r.provisioningConfig = *cfg
-
 	glog.V(4).Infof("glusterfs: creating volume with configuration %+v", r.provisioningConfig)
-	glusterfs, sizeGB, err := r.CreateVolume()
+	reqGid = gidMin + gidRandomizer.Int63n(gidMax)
```

Comment:
I feel super dense and feel like I have 2 questions I'm missing.

1. How is an admin able to control things? It seems to me the GID is some random value between 2000 and 600000. What control does an admin have over the number?
2. What prevents reuse, it seems as if even if the last PV created used say gid=101010 the next PV created could also use gid=101010. We'd hope there was only a 1/(600000-2000) chance of that happening, but that isn't great to me. I can just create and delete PVs until I get the number I was...  This is why I say it needs reservations.